corehq/warehouse/transforms/sql/location_dim.sql

```diff
@@ -0,0 +1,130 @@
+INSERT INTO {{ location_dim }} (
+    domain,
+
+    location_id,
+    sql_location_id,
+
+    level,
+    location_level_0,
+    location_level_1,
+    location_level_2,
+    location_level_3,
+    location_level_4,
+    location_level_5,
+    location_level_6,
+    location_level_7,
+
+    name,
+    site_code,
+    external_id,
+    supply_point_id,
+    is_archived,
+    latitude,
+    longitude,
+
+    location_type_name,
+    location_type_code,
+    location_type_id,
+
+    location_last_modified,
+    location_created_on,
+
+    dim_last_modified,
+    dim_created_on,
+    -- TODO: Figure out how to handle deletes when we actually hard deleted the model
+    deleted
+)
+WITH RECURSIVE location_hierarchy(
```

Comment:
> i thought you mean just joining the table to itself.

ah how is that possible without some sort of recursive syntax? you won't know how many joins to do? or are you suggesting just join 8 times?

> also does this work when we pull in a child location to the staging table but not its parent if the parent wasn't modified in the window?

going to play around with this today. in regards to doing foreign keys in a separate query, it might be, though i would imagine the query to update the foreign keys would look similar to the current one without all the meta data. will experiment